patches/mlibc/jinx-working-patch.patch

```diff
@@ -782,10 +782,10 @@ index 00000000..664c0486
 +int sys_isatty(int fd) {
 +    struct winsize ws;
 +
-+    if (!sys_ioctl(fd, TIOCGWINSZ, &ws, nullptr))
-+        return 0;
++    if (auto err = sys_ioctl(fd, TIOCGWINSZ, &ws, nullptr))
++        return err;
```

Comment:
I think this should check whether the error returned by ioctl was EBADFD or something else, in which case it should return ENOTTY. The ioctl for most files will return either ENOSYS or EINVAL, which is not what POSIX dictates for isatty

Reply:
I briefly considered that too when I opened the PR, but as long as ioctl `TIOCGWINSZ` is working properly, it shouldn’t return `ENOSYS` or `EINVAL`. I suppose there is a slight chance `EFAULT` might occur if the process stack is corrupted somehow, but at that point, the calling process has more things to worry about than the return value of isatty.

I can add an assertion that the error returned by ioctl is either `ENOTTY` or `EBADF` if you want, but other than that I’d argue that returning `ENOTTY` if the error is anything other than `EBADF` just ignores the fact that something terribly wrong (like `TIOCGWINSZ` being broken) is happening if it’s returning something like `EINVAL`.

Reply:
TIOCGWINSZ is an ioctl that's handled only by a TTY, ioctl on a file will not return ENOTTY

Reply:
Hmm, I see. In Tonix’s VFS each vnode has a `VnodeType` enum that can be `VnodeType::Terminal` and it just returns `ENOTTY` if it isn’t, but that’s kind of a clumsy way to do it. I guess we’re postponing this PR until zigux’s ioctl is fixed, then.